rfc/rfc021-vp_token-grant-type.md

```diff
@@ -132,20 +136,67 @@ The second paragraph describes the requirements that apply to JWT encoded VP's.
+
+### 4.5 Error Response
+
+If the Authorization Server determines that the VP is invalid, the Authorization Server MUST return an error response as defined in OAuth 2.0 [RFC6749].
+In addition to the error response defined in OAuth 2.0 [RFC6749], the Authorization Server MUST use the following error codes when the VP is invalid:
+
+* `invalid_verifiable_presentation`: The VP is invalid. This error code is used when the signature is incorrect or when a required field is missing.
+* `invalid_presentation_submission`: The Presentation Submission is invalid. This error code is used when the Presentation Submission is not an answer to the Presentation Definition that corresponds with the requested scope.
+* `invalid_verifiable_credentials`: The submitted Verifiable Credentials do not meet the requirements. This error code is used when the Verifiable Credentials aren't corresponding to the Presentation Definition or when the Verifiable Credentials are expired, not trusted or invalid.
 
 ## 5. Presentation Definition endpoint
 
 In order for a client to know which Presentation Definition [PE] to use, the Authorization Server MUST provide a Presentation Definition endpoint.
 The Presentation Definition endpoint MUST be registered as a `presentation_definition_endpoint` in the Authorization Server metadata [RFC8414].
-The Presentation Definition endpoint MUST return a Presentation Definition that corresponds with the requested scope.
+The Presentation Definition endpoint MUST return a Presentation Definition array that corresponds with the requested scope.
 The endpoint has a single query parameter `scope` that contains the requested scope. The parameter may contain multiple values.
+Values are separated by a space and MUST be URL encoded.
 
 The following example shows a request to the Presentation Definition endpoint:
 
-    GET /presentation_definition?scope=a&scope=b HTTP/1.1
+    GET /presentation_definition?scope=a+b HTTP/1.1
+    Host: as.example.com
+
+## 6. Scopes
+
+The scope parameter is used to indicate which Verifiable Credentials are required to access a resource.
+The OAuth 2.0 Authorization Framework allows the use of multiple scopes. 
+To simplify processing, this specification only allows a single main scope and multiple sub-scopes.
+Allowing mixing of different main scopes in a single access token request is out of scope of this specification.
+The Authorization Server MUST reject any request that contains multiple main scopes.
+
+A main scope is defined as the first scope in the scope parameter.
+It MUST be present and MUST be a string and can't contain `:`.
+A sub-scope is defined as any scope after the main scope.
+It MUST be a string and can contain `:` to indicate specific resources that need to be accessed.
+It SHOULD follow the format `resource:id:operation` where `resource` is the resource type, `id` is the specific identifier of the resource and `operation` is the operation that is allowed on the resource.
```

Comment:
Not sure, do we need to specify this? FHIR and OpenEHR scopes don't follow this exact pattern